src/librustdoc/clean/types.rs

```diff
@@ -113,6 +113,26 @@ impl Item {
         self.attrs.doc_value()
     }
 
+    /// Convenience wrapper around [`Self::from_inner`] which converts `hir_id` to a [`DefId`]
+    pub fn from_hir(hir_id: hir::HirId, inner: ItemEnum, cx: &DocContext<'_>) -> Item {
```

Comment:
```suggestion
    pub fn from_hir_id_and_kind(hir_id: hir::HirId, inner: ItemEnum, cx: &DocContext<'_>) -> Item {
```

Reply:
The naming convention in rustc is `struct Item { kind: ItemKind, ... }`, it would be great if rustdoc followed it.

So I suggest to rename `ItemEnum` -> `ItemKind` and `Item::inner` -> `Item::kind` as well, perhaps even in this PR (but not necessarily).